Libraries/LibGUI/Clipboard.cpp

```diff
@@ -62,6 +64,26 @@ Clipboard::DataAndType Clipboard::data_and_type() const
     return { data, type };
 }
 
+AK::RefPtr<Gfx::Bitmap> Clipboard::get_bitmap(Gfx::BitmapFormat format, Gfx::Size size, pid_t caller_pid)
+{
+    auto response = WindowServerConnection::the().send_sync<Messages::WindowServer::GetClipboardContents>();
+    if (response->shbuf_id() < 0)
+        return {};
+    auto shared_buffer = SharedBuffer::create_from_shbuf_id(response->shbuf_id());
+
+    if (!shared_buffer) {
+        dbgprintf("GUI::Clipboard::data() failed to attach to the shared buffer\n");
+        return {};
+    }
+    if (response->content_size() > shared_buffer->size()) {
+        dbgprintf("GUI::Clipboard::data() clipping contents size is greater than shared buffer size\n");
+        return {};
+    }
+    shared_buffer->share_with(caller_pid);
```

Comment:
No no no, if anyone, it's the WindowServer who should be calling `share_with`. And you're not going to be able to even `create_from_shbuf_id()` if the owner hasn't shared it with you.

Reply:
> And you're not going to be able to even create_from_shbuf_id() if the owner hasn't shared it with you.

If that *does* work, it's a kernel bug that needs to be fixed.

Reply:
I'll have a look at this when I have time. I don't know why that's bad ; I assumed I had to share the buffer with PaintBrush before PB could do anything with it. Maybe I can pass the data without doing that. I can't check it right now. Thank you for the info :)

Reply:
Somebody who already has access to the buffer has to share it with PB for PB to be able to access it, that much is true. But PB cannot share it with itself :slightly_smiling_face: Somebody else must share it with PB first.

Reply:
Okay I now see what I did. I'm not even using the shared buffer on PB, just interfacing with clipboard _Bitmap-wise_. So no need to share. I'll remove that. Thanks!